### ISSUE TITLE
Add image to EntryLink

### ISSUE THREAD

**Issue by toluaina** [crudlio/crudl-example-django#9]
Hello,

I would like to add an image field to the EntryLink model.
THisd is do that I can add url, title and image as links.
I keep getting "Invalid value" error when selecting the image to add.
Is there a better way of doing this.

I am using this code to add the image in entries.js. 
This needs to be hinged on the EntryLink model as a tab

```
{
     name: 'image',
     label: 'Image',
     field: 'File',
     initialValue: { value: undefined },
     normalize: image => ({ value: image.name, label: image.name, previewURL: image.url }),
     denormalize: field => field.value,
     readAs: 'DataURL', /
     onSelect: (file, dataURL) => ({
         value: { name: file.name, file: dataURL.split(',')[1] },
         label: file.name,
         previewURL: file.size < 1000000 ? dataURL : undefined,
     }),
},
```

thanks

**Comment by sehmaschine** [comment]
When exactly do you get "Invalid value"? When selecting the image or when trying to submit the form?

**Comment by sehmaschine** [comment]
Sorry, I still don´t get it ...

Are you talking about an error with CRUDL (frontend) or with Django (backend)? Or are you asking me to extend the demo to include an image field?

**Comment by toluaina** [comment]
This is an error in the CRUDL frontend. It returns an  "Invalid value" when adding a field with an image.
I have added a new field to the demo project.
So the EntryLink model now contains an image field. 
Now looks like 

```
class EntryLink(models.Model):
    entry = models.ForeignKey(Entry, verbose_name=u"Entry", related_name="links")
    url = models.URLField(u"URL", max_length=200)
    title = models.CharField(u"Title", max_length=200)
    description = models.CharField(u"Description", max_length=200, blank=True)
    position = models.PositiveIntegerField(u"Position", blank=True, null=True)
    image = models.ImageField(upload_to=image_upload_path, blank=True, null=True)

    class Meta:
        verbose_name = u"Link"
        verbose_name_plural = u"Links"
        ordering = ("entry", "title",)

    def __unicode__(self):
        return u"%s" % (self.title)
```

I would like to add this image via the front end which is when it errors.
Seems to be only a problem for image fields though.

**Comment by sehmaschine** [comment]
Did you add the new field to the API?

**Comment by toluaina** [comment]
I've added it to the schema.py as well if this is what you mean.

**Comment by sehmaschine** [comment]
I suggest to check the API first – either REST or GraphQL (I assume you're using GraphQL since you mentioned schema.py). If you then come back with more details I'm happy to look into this, but for now it´s really hard to follow this issue. Since you already worked out the setup, it should be straightforward to do some debugging and see where "Invalid value" is thrown and why.

**Comment by toluaina** [comment]
Thanks for that. Really appreciate the feedback
I'll try and be a bit clearer. Yes you're correct I'm referring to the GraphQl api.

For my use case, I would like to add a collection of components.
For example, I would like to add a list of images.

e.g in [links tab](http://crudl.io/demo/entries/25?tab=1) in the demo, you have a collection of links. I would like to do exactly the same but for images .
I have followed the same approach to creating a model and connected it up.
That is when i get "Invalid value" in the CRUDL frontend.
I noticed the image was commented out of the examples with a comment in the example
https://github.com/crudlio/crudl-example-django/blob/master/crudl-admin-rest/admin/views/entries.js

There is also a comment about this in the [schema](https://github.com/crudlio/crudl-example-django/blob/master/apps/blog/schema.py)

thanks again
